storage/src/journal/fixed.rs

```diff
@@ -153,32 +160,62 @@ impl<E: Storage + Metrics, A: Codec<Cfg = ()> + FixedSize> Journal<E, A> {
         context.register("pruned", "Number of blobs pruned", pruned.clone());
         tracked.set(blobs.len() as i64);
 
-        // truncate the last blob if it's not the expected length, which might happen from unclean
+        // Truncate the last blob if it's not the expected length, which might happen from unclean
         // shutdown.
+        let mut truncated = false;
```

Comment:
Because it is cheap to do, we check for consistency during `journal::Fixed` startup (whereas `journal::Variable` only does so during `replay()`).